src/Compilers/CSharp/Test/Semantic/Semantics/AccessCheckTests.cs

```diff
@@ -1525,5 +1525,181 @@ unsafe class A
+                {
+                    [Microsoft.CodeAnalysis.Embedded]
+                    {{innerAccessibility}} class NestedEmbedded { }
+
+                    {{innerAccessibility}} class NestedNonEmbedded { }
+                }
+                """;
+
+            var comp1 = CreateCompilation(source1, assemblyName: "Assembly1");
+            comp1.VerifyDiagnostics();
+
+            var outerSymbolInComp1 = comp1.GetTypeByMetadataName("OuterSample").GetPublicSymbol();
+            var nestedEmbeddedSymbolInComp1 = comp1.GetTypeByMetadataName("OuterSample+NestedEmbedded").GetPublicSymbol();
+            var nestedNonEmbeddedSymbolInComp1 = comp1.GetTypeByMetadataName("OuterSample+NestedNonEmbedded").GetPublicSymbol();
+
+            Assert.NotNull(outerSymbolInComp1);
+            Assert.NotNull(nestedEmbeddedSymbolInComp1);
+            Assert.NotNull(nestedNonEmbeddedSymbolInComp1);
+
+            // The types should be accessible within their own assembly
+            var assembly1SymbolPublic = comp1.Assembly.GetPublicSymbol();
+            Assert.True(((Compilation)comp1).IsSymbolAccessibleWithin(outerSymbolInComp1, assembly1SymbolPublic));
+            Assert.True(((Compilation)comp1).IsSymbolAccessibleWithin(nestedEmbeddedSymbolInComp1, assembly1SymbolPublic));
+            Assert.True(((Compilation)comp1).IsSymbolAccessibleWithin(nestedEmbeddedSymbolInComp1, outerSymbolInComp1));
+            Assert.True(((Compilation)comp1).IsSymbolAccessibleWithin(nestedNonEmbeddedSymbolInComp1, assembly1SymbolPublic));
+            Assert.True(((Compilation)comp1).IsSymbolAccessibleWithin(nestedNonEmbeddedSymbolInComp1, outerSymbolInComp1));
+
+            var comp1Reference = comp1.ToMetadataReference();
+            var comp2 = CreateCompilation("class OtherClass { }", references: [comp1Reference], assemblyName: "Assembly2");
+
+            // Get types from the referenced assembly
+            var assembly1Symbol = comp2.GetReferencedAssemblySymbol(comp1Reference);
+            var outerSymbol = comp2.GetTypeByMetadataName("OuterSample").GetPublicSymbol();
+            var nestedEmbeddedSymbol = assembly1Symbol.GetTypeByMetadataName("OuterSample+NestedEmbedded").GetPublicSymbol();
+            var nestedNonEmbeddedPublicSymbol = assembly1Symbol.GetTypeByMetadataName("OuterSample+NestedNonEmbedded").GetPublicSymbol();
+            Assert.NotNull(outerSymbol);
+            Assert.NotNull(nestedEmbeddedSymbol);
+            Assert.NotNull(nestedNonEmbeddedPublicSymbol);
+
+            var otherClassSymbol = comp2.GetTypeByMetadataName("OtherClass").GetPublicSymbol(); ;
+            var assembly2Symbol = comp2.Assembly.GetPublicSymbol();
+
+            Assert.True(((Compilation)comp2).IsSymbolAccessibleWithin(outerSymbol, otherClassSymbol));
+            Assert.True(((Compilation)comp2).IsSymbolAccessibleWithin(outerSymbol, assembly2Symbol));
+
+            // Embedded on a nested type has no effect
```

Comment:
>// Embedded on a nested type has no effect [](http://example.com/codeflow?start=12&length=42)

I am not suggesting to make a change in the current PR, but should we adjust this behavior since we expect users applying the attribute explicitly?